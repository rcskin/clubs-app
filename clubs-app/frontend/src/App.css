/* Set font color and font-family of body */
body {
    color: black;
    font-family:Georgia, 'Times New Roman', Times, serif;
  }

/* Set color and size of header */
  .header {
    background-color: rgba(223,225,220,255);
    height: 100px;
    width: 100%;
    display: flex;
    align-items: center;
    border-bottom: 2px solid black;
    position: relative;
  z-index: 1;
  }

  
  .logo {
    position: absolute;
    top: 0;
    right: 0;
    height: 100px;
    width: 200px;
    margin-right: 20px;
  }

  h1{
    padding-left: 20px;
  }

  .app-container {
    display: flex;
    justify-content: center;
    flex-direction: column;
    align-items: center;  
  }
  
  .admin-container label, 
  .admin-container input, 
  .LoginPage-container label, 
  .LoginPage-container input {
    font-size: 1.5em;
    margin-bottom: 10px;
  }
  .LoginPage-container {
    margin-top: 100px;
  }

  button {
    color: white;
    background-color: rgba(128,0,0,255);
    border-radius: 10px;
    margin-right: 5px;
    border: none;
  }
.app-container button {
    margin-bottom: 30px;
    font-size: 1.2em;
    padding: 10px 20px;
}
  .logout-btn1, .logout-btn2 {
    margin-left: 20px;
  }
  
  
  h2 {
    color: rgba(128, 0, 0, 255);
    text-align: center;
  }

  .pupil-page {
    padding-bottom: 50px;
  }

  .pupil-page .join-btn {
    margin-bottom: 5px;
  }
  
  /* Set table styles */
  table {
    border-collapse: collapse;
    width: 90%;
    margin: 0 auto; 
    table-layout: fixed;
  }
  
  td, th {
    border: 1px solid black;
    padding: 8px;
    text-align: center; /* Center the text */
  }
  
  /* Set thead background color */
  thead {
    background-color: rgba(242, 242, 242, 255);
  }
  
  /* Set tbody background color */
  tbody {
    background-color: white;
  }
  .admin-page {
    padding-bottom: 50px;
  }
  .admin-page h1 {
    color: rgba(128, 0, 0, 255);
  }
  /* Style the Add club form */
  .create-club-form {
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
    align-items: center;
    border: 1px solid black;
    padding: 10px;
  }
  
  .create-club-form div {
    margin: 10px;
  }
  
  .create-club-form label {
    display: inline-block;
    width: auto;
    text-align: right;
    margin-right: 10px;
    font-size: 1rem;
    margin-bottom: 10px;
  }
  
  .create-club-form input {
    width: auto;
    padding: 5px;
    border: 1px solid #ccc;
    border-radius: 4px;
    box-sizing: border-box;
    font-size: 1rem;
    margin-bottom: 10px;
  }

  .create-club-form button {
    margin-bottom: 10px;
    font-size: 1.2rem;
    padding: 5px 10px;
  }

  .club-table .amend-btn {
    margin-bottom: 5px;
    font-size: 1.2rem;
    padding: 5px 10px;
  }
  
  nav {
    background-color: #f2f2f2;
    padding: 10px;
  }
  
  ul {
    display: flex;
    justify-content: space-between;
    list-style-type: none;
    margin: 0;
    padding: 0;
  }
  
  li {
    margin-right: 20px;
  }
  
  li:last-child {
    margin-right: 0;
  }
  
  a {
    color: #333;
    text-decoration: none;
    font-weight: bold;
  }
  
  a:hover {
    text-decoration: underline;
  }
  