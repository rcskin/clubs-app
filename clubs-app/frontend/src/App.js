import React, { useState } from "react";
import { Routes, Route } from "react-router-dom";
import AdminDashboard from "./components/AdminDashboard";
import PupilPage from "./components/PupilPage";
import LoginPage from "./components/LoginPage";
import AdminLoginForm from "./components/AdminLoginForm";
import "./App.css";
import Header from "./components/Header";
import NavBar from "./components/NavBar";
import JoinedClubs from "./components/JoinedClubs";

const App = () => {
  // State variables to keep track of user login status and admin status
  const [isLoggedIn, setIsLoggedIn] = useState(false);
  const [isAdmin, setIsAdmin] = useState(false);

  // Function to handle login
  // Takes a boolean isAdmin parameter to determine if the user is an admin
  const handleLogin = (isAdmin) => {
    // Set isLoggedIn to true
    setIsLoggedIn(true);
    // Set isAdmin based on the provided parameter
    setIsAdmin(isAdmin);
  };

  // Function to handle logout
  const handleLogout = () => {
    // Set isLoggedIn to false
    setIsLoggedIn(false);
    // Set isAdmin to false
    setIsAdmin(false);
  };

  return (
    <div>
      {/* Render the Header component */}
      <Header />
      <div className="app-container">
        {isLoggedIn ? ( // If user is logged in
          isAdmin ? ( // If user is an admin
            <AdminDashboard handleLogout={handleLogout} /> // Render the AdminDashboard component
          ) : (
            <>
              {" "}
              {/* If user is not an admin */}
              <NavBar /> {/* Render the NavBar component */}
              <Routes>
                {" "}
                {/* Render the Routes component */}
                <Route
                  exact
                  path="/"
                  element={<PupilPage handleLogout={handleLogout} />}
                />{" "}
                {/* Render the PupilPage component */}
                <Route
                  exact
                  path="/joined-clubs"
                  element={<JoinedClubs />}
                />{" "}
                {/* Render the JoinedClubs component */}
              </Routes>
            </>
          )
        ) : (
          <>
            {" "}
            {/* If user is not logged in */}
            <LoginPage handleLogin={handleLogin} />{" "}
            {/* Render the LoginPage component */}
            <AdminLoginForm handleLogin={() => handleLogin(true)} />{" "}
            {/* Render the AdminLoginForm component */}
          </>
        )}
      </div>
    </div>
  );
};

export default App;

/* Notes:
By using the isAdmin parameter in the handleLogin function, the app can differentiate between 
regular users (pupils) and the admin user. It displays different content based on 
the users in the application
*/
