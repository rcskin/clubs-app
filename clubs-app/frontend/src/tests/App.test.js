import React from "react";
import renderer from "react-test-renderer";
import App from "../App";

test("renders App component correctly", () => {
    // Render the App component using react-test-renderer
    const tree = renderer.create(<App />).toJSON();
  
    // Assert that the rendered output matches the stored snapshot
    expect(tree).toMatchSnapshot();
  });
  