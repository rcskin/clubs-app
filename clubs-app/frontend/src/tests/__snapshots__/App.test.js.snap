// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App component correctly 1`] = `
<div>
  <div
    className="header"
  >
    <h1>
      STC Extra-Curricular Clubs App
    </h1>
    <img
      alt="STC logo"
      className="logo"
      src="STC-logo.png"
    />
  </div>
  <div
    className="app-container"
  >
     
    <div
      className="LoginPage-container"
    >
      <form
        className="LoginPage"
        onSubmit={[Function]}
      >
        <label
          htmlFor="pupil-username"
        >
          Username:
        </label>
        <input
          id="pupil-username"
          onChange={[Function]}
          required={true}
          type="text"
          value=""
        />
        <br />
        <label
          htmlFor="pupil-password"
        >
          Password:
        </label>
        <input
          id="pupil-password"
          onChange={[Function]}
          required={true}
          type="password"
          value=""
        />
        <br />
        <button
          onClick={[Function]}
          type="button"
        >
          Need to register? Click here.
        </button>
        <button
          type="submit"
        >
          Login
        </button>
      </form>
    </div>
     
    <div
      className="admin-container"
    >
      <h3>
        Admin Login
      </h3>
      <form
        className="admin"
        onSubmit={[Function]}
      >
        <label
          htmlFor="admin-username"
        >
          Admin Username:
        </label>
        <input
          id="admin-username"
          onChange={[Function]}
          required={true}
          type="text"
          value=""
        />
        <br />
        <label
          htmlFor="admin-password"
        >
          Admin Password:
        </label>
        <input
          id="admin-password"
          onChange={[Function]}
          required={true}
          type="password"
          value=""
        />
        <br />
        <button
          type="submit"
        >
          Login
        </button>
      </form>
    </div>
     
  </div>
</div>
`;
